### ISSUE TITLE
Aliasing forced on type in cast

### ISSUE THREAD

**Issue by fdw** [sqlfluff/sqlfluff#3116]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Similar to #2710, the `max` in `nvarchar(max)` is understood as a column reference and prefixed with an alias.
So, given
```sql
select cast(blah, nvarchar(max))
from table t
```
`sqlfluff` changes it to
```sql
select cast(t.blah, nvarchar(t.max))
from table t
```

### Expected Behaviour

`nvarchar(max)` is understood as a type and not changed.

### Observed Behaviour

sqlfluff produces broken SQL because the alias is prepended.

### How to reproduce

See above

### Dialect

T-SQL

### Version

Latest git `main`

### Configuration

Nothing relevant, I think

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
To me this is not "Similar to #2710" but is a direct duplicate. And the answer would be the same. Is there a reason you see it as different?

**Comment by fdw** [comment]
#2710 results in an error, but in my case, it is fixed (but broken). I wasn't sure if that is the same problem or something only similar, so I opened this issue.

Sorry for the additional work. I'll see if I can implement your suggestion 🙂

**Comment by tunetheweb** [comment]
Yeah think #270 is for `lint` and complaining about rule L028.
You ran fix and L028 "fixed" this incorrectly.
So same thing I think.

The solution should be simple enough, as suggested in #270 with a test case.

Gimme a shout if that's not the case and you need any help.

**Comment by tunetheweb** [comment]
Closed with #3130
